src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -349,6 +349,14 @@ protected override void OnHostStarted()
             _secretManager.PurgeOldFiles(Instance.ScriptConfig.RootScriptPath, Instance.TraceWriter);
         }
 
+        protected override void OnTimeout()
+        {
+            base.OnTimeout();
+            Stop();
```

Comment:
It looks like shutdownTimeout defaults to 30 seconds, which I'm definitely not seeing. I'll need to do some more investigation on this.

[https://msdn.microsoft.com/en-us/library/system.web.configuration.hostingenvironmentsection.shutdowntimeout](https://msdn.microsoft.com/en-us/library/system.web.configuration.hostingenvironmentsection.shutdowntimeout)

---
In reply to: [76060160](https://github.com/Azure/azure-webjobs-sdk-script/pull/604#discussion_r76060160) [](ancestors = 76060160,75968751)